src/slack.coffee

```diff
@@ -203,6 +203,16 @@ class SlackBot extends Adapter
     for msg in messages
       continue if msg.length < SlackBot.MIN_MESSAGE_LENGTH
 
+      # Replace @username with <@UXXXXX> for mentioning users and channels
+      msg = msg.replace /(?:^| )@([\w]+)/gm, (match, p1) =>
+        user = @client.getUserByName p1
+        if user
+          match = match.replace /@[\w]+/, "<@#{user.id}>"
+        else if (p1 is 'channel' or p1 is 'everyone' or p1 is 'group')
```

Comment:
It'd be nice to pull out the array @ line 181 and then use that here for a membership check.

(Also be sure to sync to get the merge of https://github.com/slackhq/hubot-slack/pull/218)